src/main/java/fr/insalyon/citi/golo/compiler/JavaBytecodeGenerationGoloIrVisitor.java

```diff
@@ -468,20 +495,12 @@ public void visitFunctionInvocation(FunctionInvocation functionInvocation) {
     if (functionInvocation.isAnonymous() || functionInvocation.isOnReference() || functionInvocation.isOnModuleState()) {
       methodVisitor.visitTypeInsn(CHECKCAST, "java/lang/invoke/MethodHandle");
       MethodType type = genericMethodType(functionInvocation.getArity() + 1).changeParameterType(0, MethodHandle.class);
-      visitInvocationArguments(functionInvocation);
-      methodVisitor.visitInvokeDynamicInsn(
-          functionInvocation.getName().replaceAll("\\.", "#"),
-          type.toMethodDescriptorString(),
-          CLOSURE_INVOCATION_HANDLE,
-          functionInvocation.isConstant() ? 1 : 0);
-    } else {
```

Comment:
Refactoring
the `visitInvokeDynamicInsn` statement was used twice with different arguments. Now, the arguments are initialized for a standard function invocation and overrided if and only the the conditions are met. 
The consequence is less code and only one `visitInvokeDynamicInsn`.